toolchain/check/eval.cpp

```diff
@@ -2036,19 +2041,22 @@ auto TryEvalTypedInst<SemIR::WhereExpr>(EvalContext& eval_context,
                                more_info.rewrite_constraints);
             info.other_requirements |= more_info.other_requirements;
           }
-        } else {
-          // TODO: Handle `impls` constraints beyond `.Self impls`.
-          info.other_requirements = true;
+          continue;
         }
-      } else {
-        // TODO: Handle other requirements
+
+        // TODO: Handle `impls` constraints beyond `.Self impls`.
         info.other_requirements = true;
+        continue;
       }
+
+      // TODO: Handle other requirements
```

Comment:
Note, this seems like it'd end up being structured as an `else` or `switch` from the `if` on 2018-2019 (mainly noting this due to my above comment, regarding flow). Not asking for changes, just stating the understanding as I think about the above comment.

Reply:
Yeah it is a set of if/else-if essentially. Maybe that would be more clear?

Reply:
Let me send another PR to restructure and we can discuss.